data_diff/database.py

```diff
@@ -66,12 +70,77 @@ def _query_conn(conn, sql_code: str) -> list:
     return c.fetchall()
 
 
-class Database(ABC):
+class ColType:
+    pass
+
+
+@dataclass
+class PrecisionType(ColType):
+    precision: Optional[int]
+
+
+class TemporalType(PrecisionType):
+    pass
+
+
+class Timestamp(TemporalType):
+    pass
+
+
+class TimestampTZ(TemporalType):
+    pass
+
+
+class Datetime(TemporalType):
+    pass
+
+
+@dataclass
+class UnknownColType(ColType):
```

Comment:
What do you think about before launch we create basic types for _all_ types we talked about having strong support for: ints, datetime, floats + currency, bools, and strings. Then, if you are using a column of _any_ other type we print a bunch of warnings in `_parse_type` or in `diff_tables` where we say (if the source and target are different databases):

"Column `settings` is type `json`. While it _may_ work, we haven't tested this type."

Then we could add a type support table like this to the README:

| Category          | Postgres | MySQL | Oracle | Snowflake | Redshift |
|-------------------|----------|-------|--------|-----------|----------|
| int               | ✅        | ✅     | ✅      | ✅         | ✅        |
| datetime          | ✅        | ✅     | ✅      | ✅         | ✅        |
| text              | ✅        | ✅     | ✅      | ✅         | ✅        |
| float             | ⌛        | ⌛     | ⌛      | ⌛         | ⌛        |
| decimal, currency | ⌛        | ⌛     | ⌛      | ⌛         | ⌛        |
| bool              | ⌛        | ⌛     | ⌛      | ⌛         | ⌛        |
| date, time        | ⌛        | ⌛     | ⌛      | ⌛         | ⌛        |
| json              | ⌛        | ⌛     | ⌛      | ⌛         | ⌛        |
| array             | ⌛        | ⌛     | ⌛      | ⌛         | ⌛        |

My tests will then cover each cell in that matrix against each other cell in that row.

Reply:
I think that's a lot of work, so I don't think we'll make it before launch.

We can try though.

Reply:
Ya, a nice-to-have, post-launch is fine too.